client/src/components/TagsMultiselect/HeadlessMultiselect.vue

```diff
@@ -266,13 +267,26 @@ watch(
     }
 );
 
+function getPopupLayerId() {
+    try {
+        assertDefined(root.value);
+        const popupLayerRootElement = root.value.closest("dialog, #app");
+        assertDefined(popupLayerRootElement);
+        return popupLayerRootElement.id;
+    } catch (e) {
+        // Fallback to the app layer if no dialog is found
+        return "app";
+    }
```

Comment:
```suggestion
    if (root.value) {
        const closestDialog = root.value.closest("dialog");
        return closestDialog.id ?? "app";
    } else {
        return "app";
    }
```

if we just catch the error state, without addressing it, a conditional is cleaner.

Reply:
Unfortunately, that broke the test again :sweat: 

https://github.com/galaxyproject/galaxy/actions/runs/15348141093/job/43189208083?pr=20362#step:7:21686

Reply:
Never mind, it was missing an optional chaining, it should be good now :+1: